### PR TITLE
The total amount of bitcoin that we have on the lightning network 

### PR DESCRIPTION
**Pull request by KAVAN-DESAI**

### CODE REVIEW

File: lib/model/app_model/list_channels.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          channels.map((channel) => AppChannels.fromJSON(channel)).toList();
+      return AppListChannels(channels: appChannels);
+    } else {
+      return AppListChannels();
+    }
+  }
+}
+
+class AppChannels {
+  /// Transaction identifier
```
Comment (KAVAN-DESAI) [inline lib/model/app_model/list_channels.dart:21]: source of the channel

Comment (KAVAN-DESAI) [inline lib/model/app_model/list_channels.dart:21]: source of the channel

File: lib/views/home/home_view.dart [hunk 2]
```diff
@@ -114,6 +117,26 @@ class _HomeViewState extends State<HomeView> {
...
+    }
+
+    /// because all the channnels repeat twice in list channels
+    totalChannelsAmount /= 2;
+
+    /// converting it to sat
+    totalChannelsAmount /= 1000;
+
+    /// converting it to Bitcoin
+    totalChannelsAmount *= 0.00000001;
```
Comment (vincenzopalazzo) [inline lib/views/home/home_view.dart:134]: We can keep it in satoshi? I have not enough bitcoin in my node to show a decent number :) 

In addition can be set as app setting this if the user want a Bitcoin amount or a sats amount

File: lib/views/home/home_view.dart [hunk 3]
```diff
@@ -18,9 +18,12 @@ class HomeView extends StatefulWidget {
...
   int _currentIndex = 0;
 
+  double amountSat = 0;
+
   @override
   void initState() {
     super.initState();
+    getamountMsat();
     _currentIndex = 1;
   }
```
Comment (vincenzopalazzo) [inline lib/views/home/home_view.dart:28]:
```suggestion
  void initState() async {
    super.initState();
    await getamountMsat();
    _currentIndex = 1;
  }
```

We need to await this right? or I'm missing the key idea here?

File: lib/views/home/home_view.dart [hunk 4]
```diff
@@ -114,6 +117,26 @@ class _HomeViewState extends State<HomeView> {
...
+    totalChannelsAmount /= 2;
+
+    /// converting it to sat
+    totalChannelsAmount /= 1000;
+
+    /// converting it to Bitcoin
+    totalChannelsAmount *= 0.00000001;
+    setState(() {
+      amountSat = totalChannelsAmount;
+    });
```
Comment (vincenzopalazzo) [inline lib/views/home/home_view.dart:137]: Do you think that a `setState` inside a `initState` is a good idea?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Some comments

File: lib/views/home/home_view.dart (hunk 2 above)
Comment (KAVAN-DESAI) [reply lib/views/home/home_view.dart:134]: Yes, option for choosing will be better!! But for now let me keep sats

File: lib/views/home/home_view.dart (hunk 3 above)
Comment (KAVAN-DESAI) [reply lib/views/home/home_view.dart:28]: We cannot make a initstate function async!

File: lib/views/home/home_view.dart (hunk 2 above)
Comment (vincenzopalazzo) [reply lib/views/home/home_view.dart]: >Should I open an issue for this?

Yes better keep track inside a issue

File: lib/views/home/home_view.dart [hunk 5]
```diff
@@ -55,7 +59,7 @@ class _HomeViewState extends State<HomeView> {
             height: 18,
           ),
           Text(
-            amountSat.toString(),
+            "${amountSat.toString()} sats",
```
[the remark this answers is not in the document]
Comment (KAVAN-DESAI) [reply lib/views/home/home_view.dart:62]: Sorry but we convert it to sats
```dart 
    /// converting it to sat
    totalChannelsAmount /= 1000;
```
Should It be millisats? we decided to keep in sats so???

[the remark this answers is not in the document]
Comment (KAVAN-DESAI) [reply lib/views/home/home_view.dart:62]: Oops I need to multiple by 1000 instead dividing in order to convert millisats to sats Sorry sorry

**Review by vincenzopalazzo — APPROVED** [review]

**Comment by vincenzopalazzo** [comment]
Tracking suggestion here https://github.com/dart-lightning/lndart.clnapp/issues/28

**Pull request merged**